expertsystem/amplitude/model.py

```diff
@@ -0,0 +1,349 @@
+        )
+        self.__set(particle_name, dynamics)
+        return dynamics
+
+    def __create_form_factor(self, particle_name: str) -> BlattWeisskopf:
+        meson_radius = self.__parameters.register_parameter(
+            name=f"MesonRadius_{particle_name}",
+            value=1.0,
+            fix=True,
+        )
+        return BlattWeisskopf(meson_radius)
+
+    def __set(self, particle_name: str, value: Dynamics) -> None:
+        _assert_arg_type(value, Dynamics)
+        if particle_name not in self.__particles:
+            raise KeyError(
+                f'Particle "{particle_name}" not in {ParticleCollection.__name__}'
+            )
+        self.__dynamics[particle_name] = value
+
+
+class KinematicsType(Enum):
+    Helicity = auto()
+
+
+class Kinematics:
+    def __init__(
+        self,
+        particles: ParticleCollection,
+        kinematics_type: KinematicsType = KinematicsType.Helicity,
+    ) -> None:
+        _assert_arg_type(particles, ParticleCollection)
+        _assert_arg_type(kinematics_type, KinematicsType)
+        self.__particles = particles
+        self.__initial_state: Dict[int, Particle] = dict()
+        self.__final_state: Dict[int, Particle] = dict()
+        self.__kinematics_type: KinematicsType = kinematics_type
+
+    @property
+    def initial_state(self) -> Dict[int, Particle]:
+        return self.__initial_state
+
+    @property
+    def final_state(self) -> Dict[int, Particle]:
+        return self.__final_state
+
+    @property
+    def kinematics_type(self) -> KinematicsType:
+        return self.__kinematics_type
+
+    def set_reaction(
+        self,
+        initial_state: Sequence[str],
+        final_state: Sequence[str],
+        intermediate_states: int,
+    ) -> None:
+        ini_particles = [self.__particles[name] for name in initial_state]
+        final_particles = [self.__particles[name] for name in final_state]
+        self.__initial_state = dict(enumerate(ini_particles))
+        self.__final_state = dict(
+            enumerate(
+                final_particles, start=len(initial_state) + intermediate_states
+            )
+        )
+
+    def id_to_particle(self, state_id: int) -> Particle:
+        particle = self.__initial_state.get(
+            state_id, self.__final_state.get(state_id, None)
+        )
+        if particle is None:
+            raise KeyError(f"Kinematics does not contain state ID {state_id}")
+        return particle
+
+
+class Node(ABC):
+    pass
+
+
+class AmplitudeNode(Node):
+    pass
+
+
+class DecayNode(AmplitudeNode):
+    pass
+
+
+class IntensityNode(Node):
+    pass
+
+
+@dataclass
+class SequentialAmplitude(AmplitudeNode):
+    amplitudes: List[AmplitudeNode] = field(default_factory=list)
+
+
+@dataclass
+class CoefficientAmplitude(AmplitudeNode):
+    component: str
+    magnitude: FitParameter
+    phase: FitParameter
+    amplitude: AmplitudeNode
+    prefactor: Optional[FitParameter] = None
```

Comment:
The prefactor should not be a `FitParameter` but a constant, so `float`

Reply:
e58141f2